packages/cinterop/src/commonMain/kotlin/io/realm/interop/NetworkTransport.kt

```diff
@@ -0,0 +1,43 @@
+package io.realm.interop
+
+interface NetworkTransport {
+
+    companion object {
+        const val GET = "get"
+        const val POST = "post"
+        const val PATCH = "patch"
+        const val PUT = "put"
+        const val DELETE = "delete"
+    }
+
+    val authorizationHeaderName: String?
+    val customHeaders: Map<String, String>
+
+    fun sendRequest(
```

Comment:
This API is not enough to support refreshing the access token. See https://github.com/realm/realm-java/pull/7552 for details on how this was solved in Java. We should do something similar here.

Reply:
Created https://github.com/realm/realm-kotlin/issues/450 to track this